From 73f51918ca234ed8f2b4fad863459acc437d7ff5 Mon Sep 17 00:00:00 2001
From: Johannes Agricola <johannes.agricola@work-microwave.com>
Date: Fri, 27 Aug 2021 11:17:18 +0200
Subject: [PATCH 09/14] Allowing to clear the read-only flag (e.g. for changing
 the U-Boot env):

# echo 0x400 > /sys/class/mtd/mtd4/flags
# fw_setenv serverip 172.16.11.137
# echo 0x800 > /sys/class/mtd/mtd4/flags
---
 drivers/mtd/mtdcore.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/mtdcore.c b/drivers/mtd/mtdcore.c
index 32a76b8feaa5..07f0c915b9ef 100644
--- a/drivers/mtd/mtdcore.c
+++ b/drivers/mtd/mtdcore.c
@@ -142,7 +142,22 @@ static ssize_t mtd_flags_show(struct device *dev,
 
 	return snprintf(buf, PAGE_SIZE, "0x%lx\n", (unsigned long)mtd->flags);
 }
-static DEVICE_ATTR(flags, S_IRUGO, mtd_flags_show, NULL);
+static ssize_t mtd_flags_store(struct device *dev,
+			       struct device_attribute *attr,
+			       const char *buf, size_t count)
+{
+	struct mtd_info *mtd = dev_get_drvdata(dev);
+	unsigned int flags;
+	int retval;
+
+	retval = kstrtouint(buf, 0, &flags);
+	if (retval)
+		return retval;
+
+	mtd->flags = flags;
+	return count;
+}
+static DEVICE_ATTR(flags, S_IRUGO | S_IWUSR, mtd_flags_show, mtd_flags_store);
 
 static ssize_t mtd_size_show(struct device *dev,
 		struct device_attribute *attr, char *buf)
-- 
2.25.1

